### ISSUE TITLE
Getter naming leads to weird issues

### ISSUE THREAD

**Issue by snarayanank2** [projectlombok/lombok#980]
Kudos to a great project! I've deleted 100s of lines of code :) Unfortunately, I had run into a weird problem once I incorporated Lombok.

The problem manifested itself when I was using Jackson to serialize POJOs to JSON. I had named my variable `uId` expecting it to show up as `u_id` in the resulting JSON. Turns out I was getting `uid`. This was breaking things. I initially filed an [issue with Jackson](https://github.com/FasterXML/jackson-databind/issues/1026) which has a bunch of details.

After some investigation, it turned out that there was a subtle difference in the naming of the getter generated by Lombok which was the cause of all my trouble. My boilerplate getter generated by eclipse would generate a getter:

```
  public void getuId() { return uId; } 
```

Lombok was generating:

```
  public void getUId() { return uId; }
```

This naming caused difference in JSON serialization which caused all sorts of other problems. Looks like I was hit by the small print stuff noted [here](https://projectlombok.org/features/GetterSetter.html)

I presume others would run into this as well. Any chance of changing this behavior or adding a flag to support this use case?

**Comment by henry1992007** [comment]
i encountered the same problem and used the default setter instead

**Comment by hao5ang** [comment]
Same problem with XML. I manually ignore the lombok getter method to solve this.

I don't this this is one issue of lombok. It's the design issue of FasterXML.

`    
    @Getter
    @Setter
    @JacksonXmlRootElement(localName = "Wrap")
    pulbic class Pojo {
        @JacksonXmlProperty(localName = "UId", isAttribute = true)
        private String uId;
        
        @JsonIgnore
        public String getUId() {
            return uId;
        }
    }
`

**Comment by Rawi01** [comment]
This was fixed in 1.18.24, it is now possible to set `lombok.accessors.capitalization` to change the behaviour. Related issue: #2693
